app/reducers/transactions.js

```diff
@@ -0,0 +1,33 @@
+import {
+  ADD_TRANSACTION,
+  REMOVE_TRANSACTION,
+  REMOVE_ALL_TRANSACTIONS,
+} from 'actions/transactions'
+
+function transactions(state = [], action) {
+  let newState
+
+  switch (action.type) {
+
+    case ADD_TRANSACTION:
+      newState = [...state]
+      newState.push({
+        tx: action.tx,
+        popupId: action.popupId,
+      })
+      return newState
+
+    case REMOVE_TRANSACTION:
+      newState = [...state]
+      return newState.filter(t => t.popupId !== action.popupId)
```

Comment:
filter returns a new copy of the array: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

So ```state.filter(....)``` should work out of the box and code has better performance.